src/Config.php

```diff
@@ -223,11 +223,13 @@ public function __set($name, $value)
         switch ($name) {
         case 'reportWidth' :
             // Support auto terminal width.
-            if ($value === 'auto'
-                && function_exists('shell_exec') === true
-                && preg_match('|\d+ (\d+)|', shell_exec('stty size 2>&1'), $matches) === 1
-            ) {
-                $value = (int) $matches[1];
+            if ($value === 'auto') {
+                if (function_exists('shell_exec') === true) {
+                    $dimensions = shell_exec('stty size 2>&1');
+                    if (is_string($dimensions) === true && preg_match('|\d+ (\d+)|', $dimensions, $matches) === 1) {
+                        $value = (int) $matches[1];
+                    }
+                }
             } else {
```

Comment:
Do we need any 'else' clauses here? If the `$value` is `auto` and we can't detect a sensible value, should we hard-code a fall-back width?
```suggestion
                        $value = (int) $matches[1];
                    } else {
                        $value = self::DEFAULT_REPORT_WIDTH;
                    }
                } else {
                    $value = self::DEFAULT_REPORT_WIDTH;
                }
            } else {
```

Reply:
IIRC, the default value is `80`, but yeah, might make sense having that encoded here. Still, I'd like to hear @gsherwood's opinion on it before implementing it.

Reply:
@gsherwood and me discussed this and I'll update this PR to include a default with value.